### PR TITLE
feat: add health check endpoint to CLI serve

### LINKED ISSUE #1066: cli/serve Health check endpoint. GET /health returning {"status": "pass"}.
Basic Health Check (Liveness)A basic liveness check returns a 200 OK status to signal that the python process is alive and responding.pythonfrom fastapi import FastAPI.

If the app is running, it returns status:healthy (json).  No more specific health planned at this time.  Perhaps check if we can have a running app w/o any module.serve().

```
app = FastAPI()

@app.get("/health", status_code=200)
async def liveness_check():
    return {"status": "pass"}
```

### PR DESCRIPTION
**Pull request by markstur**
# Pull Request

## Issue
Fixes #1066

## Description

Adds /health endpoint to the FastAPI server for Kubernetes liveness probes.

- /health: always returns 200 (liveness check)

Future readiness check (i.e., /ready) deferred to be done in a separate PR.

Assisted-by: IBM Bob

### Testing
- [x] Tests added to the respective file if code was changed
- [x] New code has 100% coverage if code was added
- [x] Ensure existing tests and github automation passes (a maintainer will kick off the github automation when the rest of the PR is populated)

### Attribution
- [x] AI coding assistants used

---

### Adding a new component, requirement, sampling strategy, or tool?

If your PR adds or modifies one of the types below, check the matching box. A checklist of type-specific review items will be posted as a comment.

- [ ] Component
- [ ] Requirement
- [ ] Sampling Strategy
- [ ] Tool

NOTE: Please ensure you have an issue that has been acknowledged by a core contributor and routed you to open a pull request against this repository. Otherwise, please open an issue before continuing with this pull request.

### CODE REVIEW

**Comment by markstur** [comment]
Did a little bit of import cleanup (unused or moved to top) which was not related to the PR but I think is good  to do  while touching the files vs the overhead of a nit PR.  I'm open to splitting if preferred.

File: cli/serve/app.py [hunk 1]
```diff
@@ -45,13 +45,50 @@
...
+    Returns:
+        dict: A dictionary with status "ready" when ready (HTTP 200).
+
+    Raises:
+        HTTPException: 503 status with error detail if server is not ready.
+    """
+    if _server_ready:
+        return {"status": "ready"}
+    else:
+        raise HTTPException(status_code=503, detail="Server not ready")
```
Comment (jakelorocco) [inline cli/serve/app.py]: Should this be an exception? I don't think anything on our side is actually causing failures. I think it should just be a response with the 503 status code and same detail / status message.

File: cli/serve/app.py [hunk 2]
```diff
@@ -295,5 +334,9 @@ def run_server(
         methods=["POST"],
         response_model=ChatCompletion | OpenAIErrorResponse,
     )
+
+    # Mark server as ready after route is successfully registered
+    _server_ready = True
+
     typer.echo(f"Serving {route_path} at http://{host}:{port}")
     uvicorn.run(app, host=host, port=port)
```
Comment (jakelorocco) [inline cli/serve/app.py]: Is it possible to even hit the readiness endpoint before the server is ready since the fastapi app isn't run with uvicorn until after the module is loaded and the route added?

Comment (markstur) [reply cli/serve/app.py]: Good point.  I will remove /ready for now.  That's what I get for sneaking in a bit extra.

FYI --

I cannot hit the not-ready state with the expected run_server() use and since I put _server_ready=True in run_server() it's not very usable any other way right now.  I didn't realize that even a breakpoint() would not make this work.

Technically, however, I can run the server w/o using run_server and I get healthy but not ready:
uv run python -c 'from cli.serve.app import app; import uvicorn; uvicorn.run(app, host="0.0.0.0", port="8080")' <-- but don't care about that right now.

I will add an issue to implement /ready later.  It should:

1 - handle shutdown signals for k8s!
2 - be for future use (speculative, but if we want to allow module(s) to have startup or other ready dependencies)

File: cli/serve/app.py (hunk 1 above)
Comment (markstur) [reply cli/serve/app.py]: uvicorn handles this the way you want it to

**Comment by markstur** [comment]
NOTE: I also changed the status return json to {"status": "pass"} because I learned that is IETF standard while k8s doesn't care what the string is.  pass, ok, healthy are common, but pass s most standard.

**Review by planetf1 — CHANGES_REQUESTED** [review]
The PR title and description say this adds both a liveness probe (`/health`) and a readiness probe (`/ready`), but the diff only includes `/health`. Marking as changes requested since it's incomplete relative to what's described — happy to approve once either the `/ready` endpoint is added or the PR scope is narrowed to health-only.

File: cli/serve/app.py [hunk 3]
```diff
@@ -52,6 +52,18 @@
 )
 
 
+@app.get("/health", status_code=200)
+async def health_check():
```
Comment (planetf1) [inline cli/serve/app.py]: Missing return type — `validation_exception_handler` below has one, so this reads a bit inconsistent. Easy fix:

```suggestion
async def health_check() -> dict[str, str]:
```

File: test/cli/test_serve.py [hunk 4]
```diff
@@ -45,6 +45,18 @@ def sample_request():
     )
 
 
+class TestHealthCheckEndpoint:
+    """Tests for the health check endpoint."""
+
+    def test_health_check(self):
+        """Test that /health endpoint returns 200 with correct JSON response."""
+        client = TestClient(app)
```
Comment (planetf1) [inline test/cli/test_serve.py]: Nit: the other test classes here use fixtures or class-level setup rather than constructing the client inline, but this is totally fine for a single test. Just worth a comment so the next person adding tests knows this is intentional:

```suggestion
        # /health is registered at module-load time — TestClient(app) is correct here
        client = TestClient(app)
```

File: test/cli/test_serve.py [hunk 5]
```diff
@@ -45,6 +45,18 @@ def sample_request():
...
+        response = client.get("/health")
+
+        assert response.status_code == 200
+        assert response.json() == {"status": "pass"}
```
Comment (planetf1) [inline test/cli/test_serve.py:63]: Nit: one edge case worth pinning while you're here — what does `POST /health` return? FastAPI will give 405 by default, which is the right behaviour (a misconfigured probe fails loudly rather than silently passing). A quick check:

```python
    def test_health_check_rejects_post(self):
        client = TestClient(app)
        assert client.post("/health").status_code == 405
```

File: cli/serve/app.py [hunk 6]
```diff
@@ -52,6 +52,18 @@
 )
 
 
+@app.get("/health", status_code=200)
```
Comment (planetf1) [inline cli/serve/app.py]: FYI: `status_code=200` is FastAPI's default for GET so it's a no-op — the code works correctly either way. Just flagging in case you want to trim it:

```suggestion
@app.get("/health")
```

File: cli/serve/app.py (hunk 2 above)
Comment (jakelorocco) [reply cli/serve/app.py]: Definitely agree that this is a good improvement to have. I think it just requires changing how the server setup happens so that the server can start up even while the things it relies on (the chat endpoint) is still loading.

File: test/cli/test_serve.py (hunk 4 above)
Comment (markstur) [reply test/cli/test_serve.py]: there is a 2nd test now (thanks), so I added a fixture and this suggested comment is in the fixture

**Review by planetf1 — APPROVED** [review]
LGTM — all concerns addressed: PR scope narrowed to health-only (readiness deferred), return type added, POST 405 test added, fixture used. Thanks for the quick turnaround.
